platform/ruby/mail_checker.rb

```diff
@@ -0,0 +1,35 @@
+                map(&:downcase).
+                to_set
+
+  def self.test(email)
+    return false unless valid_email?(email)
+
+    !BLACKLIST.include?(extract_domain(email))
+  end
+
+  def self.valid_email?(email)
+    return false if email.nil?
+
+    email =~ email_regex
+  end
+
+  # Based on PHP FILTER_VALIDATE_EMAIL
+  # See https://fightingforalostcause.net/content/misc/2006/compare-email-regex.php
+  def self.email_regex
+    /^(?!(?:(?:\x22?\x5C[\x00-\x7E]\x22?)|(?:\x22?[^\x5C\x22]\x22?)){255,})(?!(?:(?:\x22?\x5C[\x00-\x7E]\x22?)|(?:\x22?[^\x5C\x22]\x22?)){65,}@)(?:(?:[\x21\x23-\x27\x2A\x2B\x2D\x2F-\x39\x3D\x3F\x5E-\x7E]+)|(?:\x22(?:[\x01-\x08\x0B\x0C\x0E-\x1F\x21\x23-\x5B\x5D-\x7F]|(?:\x5C[\x00-\x7F]))*\x22))(?:\.(?:(?:[\x21\x23-\x27\x2A\x2B\x2D\x2F-\x39\x3D\x3F\x5E-\x7E]+)|(?:\x22(?:[\x01-\x08\x0B\x0C\x0E-\x1F\x21\x23-\x5B\x5D-\x7F]|(?:\x5C[\x00-\x7F]))*\x22)))*@(?:(?:(?!.*[^.]{64,})(?:(?:(?:xn--)?[a-z0-9]+(?:-[a-z0-9]+)*\.){1,126}){1,}(?:(?:[a-z][a-z0-9]*)|(?:(?:xn--)[a-z0-9]+))(?:-[a-z0-9]+)*)|(?:\[(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){7})|(?:(?!(?:.*[a-f0-9][:\]]){7,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?)))|(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){5}:)|(?:(?!(?:.*[a-f0-9]:){5,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3}:)?)))?(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))(?:\.(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))){3}))\]))$/i
+  end
+
+  def self.extract_domain(email)
+    domain = email.gsub(/.+@([^.]+)/, '\1').downcase
+    domain.split('.')[-2,2].join('.') # Don't include subdomains
+  end
+end
+
+def MailChecker(email)
```

Comment:
I don't like the method name being `MailChecker`, meaning that you will call this as `MailChecker('someone@example.com')`, which is anything but idiomatic Ruby.. But I opted to follow the style of the node & PHP implementation for now.

What I rather would like to do is rename `MailCheckerImpl` to `MailChecker` and then call it like this `MailChecker.test('someone@example.com')`.

@FGRibreau any thoughts?